bpf/lib/srv6.h

```diff
@@ -126,13 +126,19 @@ srv6_encapsulation(struct __ctx_buff *ctx, int growth, __u16 new_payload_len,
 		   __u8 nexthdr, union v6addr *saddr, struct in6_addr *sid)
 {
 	__u32 len = sizeof(struct ipv6hdr) - 2 * sizeof(struct in6_addr);
+	__u32 packet_hash;
+
 	struct ipv6hdr new_ip6 = {
 		.version     = 0x6,
 		.payload_len = bpf_htons(new_payload_len),
 		.nexthdr     = nexthdr,
 		.hop_limit   = IPDEFTTL,
 	};
 
+	packet_hash = get_hash_recalc(ctx);
+	new_ip6.flow_lbl[1] = (packet_hash >> 16) & 0xff;
+	new_ip6.flow_lbl[2] = (packet_hash >> 24) & 0xff;
```

Comment:
Flow label is 20bits, but we cut off 4bits here. Let's use the full range. Here's an example of the Linux's encoding. We can ignore the tc-class part.

https://github.com/torvalds/linux/blob/master/include/net/ipv6.h#L1037-L1041

Reply:
20lsb bits of hash used to populate flow label. commit description changed accordingly.